### PR TITLE
Added pow data table to ui

### PR DESCRIPTION
**Pull request by Sirmorrison**
Added pow data table the project ui, implemented pagination also or the ui.

image

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -116,7 +117,8 @@ func _main(ctx context.Context) error {
 				log.Error(err)
 			}
 		}
-		if !cfg.DisableExchangeTicks {
+
+		if cfg.DisableExchangeTicks {
```
Comment (ademuanthony) [inline main.go]: restore `if !cfg.DisableExchangeTicks {` please.
Be very careful when conducting a rebase. Do make sure you thoroughly understand the source code

File: main.go [hunk 2]
```diff
@@ -145,7 +147,7 @@ func _main(ctx context.Context) error {
 			}
 		}
 
-		if !cfg.DisablePow {
+		if cfg.DisablePow {
```
Comment (ademuanthony) [inline main.go]: restore `if !cfg.DisablePow {`

File: postgres/models/pow_data.go [hunk 3]
```diff
@@ -129,7 +129,7 @@ func (w whereHelpernull_Float64) GTE(x null.Float64) qm.QueryMod {
 }
 
 var PowDatumWhere = struct {
-	Time              whereHelperint
+	Time              whereHelpertime_Time
```
Comment (ademuanthony) [inline postgres/models/pow_data.go]: you are not supposed to modify this file. Please undo the changes

File: postgres/models/pow_data.go [hunk 4]
```diff
@@ -138,7 +138,7 @@ var PowDatumWhere = struct {
 	BTCPrice          whereHelpernull_String
 	Source            whereHelperstring
 }{
-	Time:              whereHelperint{field: "\"pow_data\".\"time\""},
+	Time:              whereHelpertime_Time{field: "\"pow_data\".\"time\""},
```
Comment (ademuanthony) [inline postgres/models/pow_data.go]: Please undo

File: web/handlers.go [hunk 5]
```diff
@@ -30,6 +30,9 @@ func (s *Server) GetExchangeTicks(res http.ResponseWriter, req *http.Request) {
 	}
 
 	totalCount, err := s.db.AllExchangeTicksCount(ctx)
+	if err != nil {
+		panic(err)
```
Comment (ademuanthony) [inline web/handlers.go]: we need to remove this `panic` when an error page is added. please add a `// todo: ` comment to this effect

File: web/handlers.go [hunk 6]
```diff
@@ -66,6 +69,9 @@ func (s *Server) GetVspTicks(res http.ResponseWriter, req *http.Request) {
 	}
 
 	totalCount, err := s.db.AllVSPTickCount(ctx)
+	if err != nil {
+		panic(err)
```
Comment (ademuanthony) [inline web/handlers.go]: we need to remove this `panic` when an error page is added. please add a `// todo: ` comment to this effect

File: web/handlers.go [hunk 7]
```diff
@@ -81,3 +87,42 @@ func (s *Server) GetVspTicks(res http.ResponseWriter, req *http.Request) {
...
+func (s *Server) GetPowData(res http.ResponseWriter, req *http.Request) {
+	req.ParseForm()
+	page := req.FormValue("page")
+
+	pageToLoad, err := strconv.ParseInt(page, 10, 32)
+	if err != nil || pageToLoad <= 0 {
+		pageToLoad = 1
+	}
+
+	var txPerPage int = recordsPerPage
```
Comment (ademuanthony) [inline web/handlers.go]: you don't need this variable

File: web/handlers.go [hunk 8]
```diff
@@ -81,3 +87,42 @@ func (s *Server) GetVspTicks(res http.ResponseWriter, req *http.Request) {
...
+	offset := (int(pageToLoad) - 1) * txPerPage
```
Comment (ademuanthony) [inline web/handlers.go]: you can use `recordsPerPage` directly in place of `txPerPerPage`

File: web/handlers.go [hunk 9]
```diff
@@ -81,3 +87,42 @@ func (s *Server) GetVspTicks(res http.ResponseWriter, req *http.Request) {
...
+
+	ctx := context.Background()
+
+	allPowDataSlice, err := s.db.FetchPowData(context.Background(), offset, recordsPerPage)
```
Comment (ademuanthony) [inline web/handlers.go]: let's replace `context.Background()` with the `ctx` define above

File: web/public/js/jquery.min.js [hunk 10]
```diff
@@ -1,2 +1,2 @@
...
...tion")),ot.type="checkbox",y.checkOn=""!==ot.value,y.optSelected=at.selected,(ot=E.createElement("input")).value="t",ot.type="radio",y.radioValue="t"===ot.value;var ht,gt=k.expr.attrHandle;k.fn.extend({attr:function(e,t){return _(this,k.attr,e,t,1<arguments.length)},removeAttr:function(e){return this.each(function(){k.removeAttr(this,e)})}}),k.extend({attr:function(e,t,n){var r,i,o=e.nodeType;if(3!==o&&8!==o&&2!==o)return"undefined"==typeof e.getAttribute?k.prop(e,t,n):(1===o&&k.isXMLDoc(e)||(i=k.attrHooks[t.toLowerCase()]||(k.expr.match.bool.test(t)?ht:void 0)),void 0!==n?null===n?void k.removeAttr(e,t):i&&"set"in i&&void 0!==(r=i.set(e,n,t))?r:(e.setAttribute(t,n+""),n):i&&"get"in i&&null!==(r=i.get(e,t))?r:null==(r=k.find.attr(e,t))?void 0:r)},attrHooks:{type:{set:function(e,t){if(!y.radioValue&&"radio"===t&&A(e,"input")){var n=e.value;return e.setAttribute("type",t),n&&(e.value=n),t}}}},removeAttr:function(e,t){var n,r=0,i=t&&t.match(R);if(i&&1===e.nodeType)while(n=i[r++])e.removeAttribute(n)}}),ht={set:function(e,t,n){return!1===t?k.removeAttr(e,n):e.setAttribute(n,n),n}},k.each(k.expr.match.bool.source.match(/\w+/g),function(e,t){var a=gt[t]||k.find.attr;gt[t]=function(e,t,n)
```
Comment (ademuanthony) [inline web/public/js/jquery.min.js]: why is this modified, I am just curious

File: web/public/js/send.js [hunk 11]
```diff
@@ -0,0 +1,307 @@
+/**==================================================================*
```
Comment (ademuanthony) [inline web/public/js/send.js]: Please delete this file

File: web/server.go [hunk 12]
```diff
@@ -72,7 +72,8 @@ func (s *Server) loadTemplates() {
 	layout := "web/views/layout.html"
 	tpls := map[string]string{
 		"exchange.html": "web/views/exchange.html",
-		"vsp.html":      "web/views/vsp.html",
+		"vsp.html": "web/views/vsp.html",
```
Comment (ademuanthony) [inline web/server.go]: `go fmt ./...`

File: web/public/js/jquery.min.js (hunk 10 above)
Comment (Sirmorrison) [reply web/public/js/jquery.min.js]: updated version

**Comment by Sirmorrison** [comment]
> 
> 
> Please make the column headers readable; e.g. `BTCPrice` should be `BTC Price`

image
